zkchannels-contract/zkchannel_smartpy_script.py

```diff
@@ -14,193 +14,240 @@
         sp.verify(self.data.merchAddr == sp.sender)
+        # Verify that the contract status is OPEN.
         sp.verify(self.data.status == OPEN)
+        # Record the time when the delay period will expire with self.data.delayExpiry.
+        # The time is calculated by adding the number of seconds in selfDelay to timestamp of
+        # the block it is confirmed in.
         self.data.delayExpiry = sp.now.add_seconds(self.data.selfDelay)
         self.data.status = EXPIRY
  
-    # merchClaim can be called by the merchant if the customer has not called
-    # custClose before the delay period has expired.
+    # merchClaim can only be called by the merchant after the expiry entrypoint has been called and # the delay period defined by selfDelay has elapsed. If the customer calls custClose, merchClaim
+    # cannot be called.
     @sp.entry_point
     def merchClaim(self):
+        # Only allow the merchant to call the entrypoint.
         sp.verify(self.data.merchAddr == sp.sender)
+        # Verify that the contract status is EXPIRY.
         sp.verify(self.data.status == EXPIRY)
-        sp.verify(self.data.delayExpiry < sp.now)
+        # Verify that the delay period has passed.
+        sp.verify(sp.now > self.data.delayExpiry)
+        # Send the total balance to the merchant.
         sp.send(self.data.merchAddr, self.data.custBal + self.data.merchBal)
-        self.data.custBal = sp.tez(0)
-        self.data.merchBal = sp.tez(0)
         self.data.status = CLOSED
 
+    # custClose can be called by the customer to initiate a unilateral channel closure. It can only 
+    # be called by the customer while the channel status is OPEN or EXPIRY. The entrypoint 
+    # validates the merchant's blind signature over a tuple containing the channel ID, close flag, 
+    # rev lock, customer closing balance and merchant closing balance.
     @sp.entry_point
     def custClose(self, custBal, merchBal, revLock, s1, s2):
+        # Only allow the customer to call the entrypoint.
         sp.verify(self.data.custAddr == sp.sender)
+        # Verify that the contract status is either OPEN or EXPIRY.
         sp.verify((self.data.status == OPEN) | (self.data.status == EXPIRY))
-        # Fail if s1 is set to 0
-        sp.verify(self.is_g1_not_zero(s1))
+
+        ## Start of PS signature verification 
+        # Fail if s1 is set to the identity element.
+        sp.verify(self.is_g1_identity(s1)==False)
         # Prepare pairing check inputs
         g2 = self.data.g2
-        y2s0 = self.data.merchPk0
-        y2s1 = self.data.merchPk1
-        y2s2 = self.data.merchPk2
-        y2s3 = self.data.merchPk3
-        y2s4 = self.data.merchPk4
-        x2 = self.data.merchPk5
+        y2s0 = self.data.y2s_0
+        y2s1 = self.data.y2s_1
+        y2s2 = self.data.y2s_2
+        y2s3 = self.data.y2s_3
+        y2s4 = self.data.y2s_4
+        x2 = self.data.x2
         cid = self.data.cid
         close_b = self.data.close_flag
```

Comment:
I believe so. I'm reluctant to anything in this part of the code because Ayo wrote it and it might still be helpful for him to remember the type conversions going on.